bundles/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/handler/SagerCasterHandler.java

```diff
@@ -0,0 +1,269 @@
+                    new Thread(() -> {
+                        sagerWeatherCaster.setRaining(isRaining == OnOffType.ON);
+                        postNewForecast();
+                    }).start();
+                    break;
+                case CHANNEL_WIND_SPEED:
+                    logger.debug("Updated wind speed, updating forecast");
+                    DecimalType newValue = (DecimalType) command;
+                    new Thread(() -> {
+                        sagerWeatherCaster.setBeaufort(newValue.intValue());
+                        postNewForecast();
+                    }).start();
+                    break;
+                case CHANNEL_PRESSURE:
+                    logger.debug("Sea-level pressure updated, updating forecast");
+                    if (command instanceof QuantityType) {
+                        @SuppressWarnings("unchecked")
+                        QuantityType<Pressure> newPressure = ((QuantityType<Pressure>) command)
+                                .toUnit(HECTO(SIUnits.PASCAL));
+                        if (newPressure != null) {
+                            pressureCache.put(newPressure);
+                            QuantityType<Pressure> agedPressure = pressureCache.getAgedValue();
+                            if (agedPressure != null) {
+                                new Thread(() -> {
+                                    sagerWeatherCaster.setPressure(newPressure.doubleValue(),
+                                            agedPressure.doubleValue());
+                                    updateState(CHANNEL_PRESSURETREND,
+                                            new StringType(String.valueOf(sagerWeatherCaster.getPressureEvolution())));
+                                    postNewForecast();
+                                }).start();
+                            } else {
+                                updateState(CHANNEL_FORECAST, FORECAST_PENDING);
+                            }
+                        }
+                    }
+                    break;
+                case CHANNEL_TEMPERATURE:
+                    logger.debug("Temperature updated");
+                    if (command instanceof QuantityType) {
+                        @SuppressWarnings("unchecked")
+                        QuantityType<Temperature> newTemperature = ((QuantityType<Temperature>) command)
+                                .toUnit(SIUnits.CELSIUS);
+                        if (newTemperature != null) {
+                            temperatureCache.put(newTemperature);
+                            currentTemp = newTemperature.intValue();
+                            QuantityType<Temperature> agedTemperature = temperatureCache.getAgedValue();
+                            if (agedTemperature != null) {
+                                // TODO
+                                // https://www.sciencedirect.com/topics/engineering/outdoor-air-temperature
```

Comment:
can this be implemented?

Reply:
Woops, don't remember what I had in mind here right now. :blush: